custom_components/adaptive_lighting/switch.py

```diff
@@ -239,53 +239,69 @@ def _split_service_data(service_data, adapt_brightness, adapt_color):
     return service_datas
 
 
-def _find_switch_with_any_of_lights(
-    hass: HomeAssistant,
-    lights: list[str],
-    service_call: ServiceCall,
-) -> AdaptiveSwitch:
-    """Find the switch that controls the lights in 'lights'."""
+def _get_switches_with_lights(
+    hass: HomeAssistant, lights: list[str], is_on: bool
+) -> list[AdaptiveSwitch]:
     config_entries = hass.config_entries.async_entries(DOMAIN)
     data = hass.data[DOMAIN]
-    switches = {}
+    switches = []
     for config in config_entries:
-        # this check is necessary as there seems to always be an extra config
-        # entry that doesn't contain any data. I believe this happens when the
-        # integration exists, but is disabled by the user in HASS.
         if config.entry_id in data:
             switch = data[config.entry_id]["instance"]
+            if is_on and not switch.is_on:
+                continue
```

Comment:
Why is this necessary?

Reply:
To use this function to extract a switch that is on. This might be useful in the case where a light appears in multiple switches, however, currently only one of those switches is on.

Reply:
Shouldn't there be another check where `switches` is empty where the only switch containing a light is off?

Reply:
I did something very similar now. Thanks for the suggestion!